### PR TITLE
Tweaks force flag to not apply to empty sites

### PR DESCRIPTION
**Pull request by skyluc**
With this, the `mvn -Ppackage` part of the build doesn't fail.

### CODE REVIEW

File: src/main/scala/org/scalaide/buildtools/EcosystemBuild.scala [hunk 1]
```diff
@@ -38,8 +37,12 @@ object EcosystemBuild {
...
   }
 
-  private def shouldBeRegenerated(baseScalaIDEVersions: Seq[ScalaIDEVersion], siteScalaIDEVersions: Set[InstallableUnit]): Boolean = {
-    if (!baseScalaIDEVersions.forall(_.associatedAvailableAddOns.isEmpty)) {
+  // the ecosystem creation can be forced via the --force=<ecosystemId> CLI argument
+  private def shouldBeRegenerated(baseScalaIDEVersions: Seq[ScalaIDEVersion], siteScalaIDEVersions: Set[InstallableUnit], forceEcosystemCreation: Boolean): Boolean = {
+    if (forceEcosystemCreation && !baseScalaIDEVersions.isEmpty) {
+      // force rebuild only if there is anything to build
+      true
+    } else if (!baseScalaIDEVersions.forall(_.associatedAvailableAddOns.isEmpty)) {
```
Comment (dotta) [inline src/main/scala/org/scalaide/buildtools/EcosystemBuild.scala]: If you have a good name for the condition (here and the above one), I'd value having two methods and refactoring the logic here as 

```
if(method1() || method2()) true
else ...
```

If you don't have a good name to assign to the conditions, then it's all good as it is.

**Comment by dotta** [comment]
LGTM

**Comment by skyluc** [comment]
New and improved flag computation.

**Comment by skyluc** [comment]
Cleaning up some old stuff :)

**Comment by dotta** [comment]
Hey @skyluc, hope all is well :)

**Pull request closed without merging**